### PR TITLE
Set `usedforsecurity=False` in hashlib methods (FIPS compliance)

### PR DESCRIPTION
**Pull request by Wauplin**
Solves https://github.com/huggingface/transformers/issues/27034 (cc @DueViktor).

This PR makes `transformers` FIPS-compliant regarding hashlib usage by setting `usedforsecurity=False` in every hashlib method (used for file checking, not cryptography purposes). It's based on utilities added in https://github.com/huggingface/huggingface_hub/pull/1782 and released in `huggingface_hub==v0.19.0`. 

**Note:** before merging this we need to release a new tokenizers version that would allow new `huggingface_hub` version (see https://github.com/huggingface/tokenizers/pull/1385). Tests are currently failing because of this which is expected. cc @ArthurZucker what's the status on the next release?

### CODE REVIEW

**Review by amyeroberts — APPROVED** [review]
Thanks for updating! 

Happy to merge once all dependencies have been handled. Would be good to have a quick review from @ydshieh who is the packages and CI master 👑

**Review by ydshieh — APPROVED** [review]
LGTM.

Usually we don't touch any file udner `examples/research_projects`, but it's fine here (for me).

(not sure the name `insecure_hashlib` is the best name, but that is not the point of this PR 😄 )

**Review by ArthurZucker — COMMENTED** [review]
Thanks! Not 100% sure we need to rely on the utility as it's a pretty small piece of code for something pretty code + not very intuitive to me that we need to import from hf hub. If others are fine good for me as well

**Review by ArthurZucker — APPROVED** [review]

File: setup.py
Comment (Wauplin) [reply setup.py:178]: Was about to post a message :smile: 
Ok then I'll wait until it's merged and update this PR. Just let me know when it's done :)

**Comment by Wauplin** [comment]
Thanks for merging https://github.com/huggingface/transformers/pull/27494 @ArthurZucker ! I merged `main` into this PR and once CI is green I think we'll be good to merge :)

**Comment by Wauplin** [comment]
Still some issue with the dependencies... :confused: 
[I opened a PR](https://github.com/huggingface/huggingface_hub/pull/1828) in huggingface_hub and will make another patch release later.

**Comment by Wauplin** [comment]
I converted this PR to draft because of the dependency issues we're having. I'll push a new fix in `huggingface_hub` to  definitely settle this (see [slack thread](https://huggingface.slack.com/archives/C014N4749J9/p1700069962026839) -internal- for explanations).

**Please do not merged even if CI is green.**

**Review by amyeroberts — APPROVED** [review]
Thanks for all the investigative work on this!

**Comment by amyeroberts** [comment]
@Wauplin I can see the PR was marked as ready for review again. Does this mean it's now mergeable? 

Would be good to get a final 👍 from @ydshieh to make sure all relevant files for our CI have been updated.

**Comment by ydshieh** [comment]
Look good!

**Comment by amyeroberts** [comment]
Confirmed with @Wauplin over slack that the PR is good to go - merging!

**Pull request merged**
